tests/models/decoder_only/vision_language/test_models.py

```diff
@@ -0,0 +1,629 @@
+"""Common tests for testing .generate() functionality for single / multiple
+image, embedding, and video support for different VLMs in vLLM.
+"""
+import os
+from pathlib import PosixPath
+from typing import Optional, Type
+
+import pytest
+from transformers import AutoModelForVision2Seq
+
+from vllm.utils import cuda_device_count_stateless, identity, is_cpu, is_hip
+
+from ....conftest import (IMAGE_ASSETS, HfRunner, VllmRunner, _ImageAssets,
+                          _VideoAssets)
+from ....utils import fork_new_process_for_each_test, get_memory_gb
+from ...utils import check_outputs_equal
+from .vlm_utils import custom_inputs, model_utils
+from .vlm_utils.case_filtering import get_parametrized_options
+from .vlm_utils.core_utils import (run_custom_inputs_test, run_embedding_test,
```

Comment:
Maybe rename the file to `runners.py`?

Reply:
Sure - I split the runners -> `runners.py` and also moved the builders in `core.py` -> `builders.py`. Then, adding a new modality will mostly entail:
- Writing a new runner
- Writing a new builder
- Adding defaults (if needed) to the types
- Writing the test wrapper

It also keeps the stuff in `core.py`, which is more extensible if it gets more complex in the future, e.g., to add support for comparing to quantized models etc